<?xml version='1.0' encoding='utf-8'?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="panda">
  <xacro:arg name="arm_id" default="panda"/> <!-- Name of this panda -->
  <xacro:arg name="hand" default="false"/> <!-- Should a franka_gripper be mounted at the flange?" -->
  <xacro:arg name="robot_ip" default=""/> <!-- IP address or hostname of the robot" -->
  <xacro:arg name="use_fake_hardware" default="false"/>
  <xacro:arg name="fake_sensor_commands" default="false"/>

  <link name="world"></link>
  <link name="panda_link8_aligned"></link>
  <link name="pen_link">
  </link>
  <joint name="end_joint" type="fixed">
    <origin rpy="0 0 -0.785" xyz="0 0 0"/>
    <parent link="panda_link8"/>
    <child link="panda_link8_aligned"/>
  </joint>

  <joint name="pen_joint" type="fixed">
    <origin rpy="0 1.5708 0" xyz="0.058 0 0.094"/>
    <parent link="panda_link8_aligned"/>
    <child link="pen_link"/>
  </joint>
  
  <link name="drawing_frame_link">
    <visual>
      <geometry>
        <mesh
          filename="package://ergodic_sketching_ros/descriptions/meshes/drawing_paper.stl"/>
      </geometry>
      <material name="grey"/>
    </visual>
  </link>

  <joint name="drawing_frame_joint" type="fixed">
    <parent link="world"/>
    <child link="drawing_frame_link"/>

    <origin rpy="0.00448539 -0.0175784  -2.47236401" xyz="0.58992866 0.19507239 0.00578767"/>
  </joint>

  <xacro:include filename="$(find franka_description)/robots/common/franka_arm.xacro"/>
  <xacro:franka_arm arm_id="$(arg arm_id)" safety_distance="0.03" connected_to="world"
    joint_limits="${xacro.load_yaml('$(find franka_description)/robots/fr3/joint_limits.yaml')}" kinematics="${xacro.load_yaml('$(find franka_description)/robots/fr3/kinematics.yaml')}" inertials="${xacro.load_yaml('$(find franka_description)/robots/fr3/inertials.yaml')}" dynamics="${xacro.load_yaml('$(find franka_description)/robots/fr3/dynamics.yaml')}"/>

  <xacro:if value="$(arg hand)">
    <xacro:include filename="$(find franka_description)/end_effectors/franka_hand/franka_hand.urdf.xacro"/>
    <xacro:hand ns="$(arg arm_id)" rpy="0 0 ${-pi/4}" connected_to="$(arg arm_id)_link8" safety_distance="0.03"/>
  </xacro:if>
  <xacro:include filename="$(find franka_description)/robots/common/franka_arm.ros2_control.xacro"/>
  <xacro:franka_arm_ros2_control arm_id="$(arg arm_id)" robot_ip="$(arg robot_ip)" use_fake_hardware="$(arg use_fake_hardware)" fake_sensor_commands="$(arg fake_sensor_commands)"/>
</robot>
